StudyHub/StudyHub.BLL/Services/AssignmentTaskService.cs

```diff
@@ -43,4 +42,46 @@ public async Task<AssignmentTaskDTO> AddTask(CreateAssignmentTaskDTO dto)
 
         return result;
     }
+
+    public async Task<bool> DeleteAssignmentTask(Guid assignmentTaskId)
+    {
+        var entity = await _assignmentTaskRepository
+            .Include(assignmentTask => assignmentTask.Options)
+            .FirstOrDefaultAsync(assignmentTask => assignmentTask.Id == assignmentTaskId);
+
+        if (entity == null)
+            throw new KeyNotFoundException($"Unable to find entity with such key {assignmentTaskId}");
+
+        return await _assignmentTaskRepository.DeleteAsync(entity);
+    }
+
+    public async Task<List<AssignmentTaskDTO>> GetAssignmentTask(Guid assignmentId)
+    {
+        var result = await _assignmentTaskRepository
+            .Include(assignmentTask => assignmentTask.Options)
+            .Where(assignmentTask => assignmentTask.AssignmentId == assignmentId)
+            .ToListAsync();
+
+        if (result.Count < 1)
+            throw new NotFoundException($"Unable to find entity with such key {assignmentId}");
+
+        return _mapper.Map<List<AssignmentTaskDTO>>(result);
+    }
+
+    public async Task<AssignmentTaskDTO> UpdateAssignmentTask(Guid assignmentTaskId, UpdateAssignmentTaskDTO dto)
+    {
+        var entity = await _assignmentTaskRepository
+            .Include(assignmentTask => assignmentTask.Options)
+            .FirstOrDefaultAsync(assignmentTask => assignmentTask.Id == assignmentTaskId);
+
+        if (entity == null)
+            throw new NotFoundException($"Unable to find entity with such key {assignmentTaskId}");
+
+        _mapper.Map(dto, entity);
+
+        await _assignmentTaskRepository.UpdateAsync(entity);
+        await _optionsService.UpdateAssignmentTaskOption(entity.Id, dto.Options);
```

Comment:
Maybe better that this function will have in parameters the whole dto, not only dto.Options?

Reply:
Better to say that it is OptionService and has to work with Options, neither with AssignmentTasks